modules/dxp/apps/scim/scim-configuration-web/src/main/java/com/liferay/scim/configuration/web/internal/portlet/action/SaveScimConfigurationMVCActionCommand.java

```diff
@@ -155,6 +169,48 @@ else if (Objects.equals(cmd, "reset")) {
 
 			_oAuth2ApplicationLocalService.deleteOAuth2Application(
 				oAuth2Application);
+
+			Indexer<User> userIndexer = IndexerRegistryUtil.nullSafeGetIndexer(
+				User.class);
+
+			Indexer<UserGroup> userGroupIndexer =
+				IndexerRegistryUtil.nullSafeGetIndexer(UserGroup.class);
```

Comment:
We should only create these `Indexer<>` objects as needed.  With the current logic, you will create two but only use one at most.

Reply:
we can have both users and user groups related to a scim client. The configurations process are separated from each other as well as the APIs